### ISSUE TITLE
[Bug] Config::validate の検証漏れ (num_hidden_layers / intermediate_size / local_attention / layer_norm_eps)

### ISSUE THREAD

**Issue by thkt** [thkt/rurico#226]
## 問題

`Config::validate` (src/modernbert/config.rs:50) の doc は「Validate invariants (non-zero sizes, divisibility)」「Returns an opaque validation message if any required size is zero」と宣言するが、以下の4項目を検証しない。

| フィールド | 検証状況 | 不正値を通した場合の挙動 |
| --- | --- | --- |
| `num_hidden_layers` | 検証なし | model.rs:236 の `(0..config.num_hidden_layers)` で layers が空になり、forward は embedding → final_norm のみで「動く」(silent degradation) |
| `intermediate_size` | i32 bound のみ (config.rs:74)、0 チェックなし | model.rs:114 で Wi が `[hidden, 0]` の Linear になる |
| `local_attention` | 上限のみ (config.rs:83-88)、0 チェックなし | model.rs:249 で `local_attention / 2 == 0` → get_local_attention_mask (model.rs:468-473) が対角のみ 0.0 の mask を生成し、各トークンが自分自身しか見えない degenerate attention のまま「動く」(silent degradation) |
| `layer_norm_eps` | 検証なし | model.rs:400 で `as f32` キャスト。負や NaN は LayerNorm 内の sqrt で NaN を生み、最終的に出力の is_finite 検証で NonFiniteOutput になるが、根本原因 (config 破損) がエラーメッセージから見えない |

## なぜ問題か

- doc は「any required size is zero」を拒否すると宣言しており、`num_hidden_layers` / `intermediate_size` は required size そのもの。doc と実装が乖離している。
- `VerifiedArtifacts` は「config.json parses as a valid Config」を保証する型として downstream / AI agent に提示される (アウトカム Behavior「AI agent が公開 API を rustdoc と型シグネチャだけで誤用なく利用できる」)。validate を通った config が空 layers や対角 mask で silent に「動いてしまう」と、埋め込み品質の劣化が型でもエラーでも検出できない。
- ruri-v3 ピン済み config では発生しないため実害は現状ゼロ。ただし `CandidateArtifacts::verify` はローカルの任意 config.json を受けるため、config を手で加工した場合や将来のモデル追加時に検証網をすり抜ける。

## 修正案

validate に以下を追加する (既存の関連フィールドごとの検証順に合わせて配置):

```rust
if self.num_hidden_layers == 0 {
    return Err("num_hidden_layers must be > 0".into());
}
if self.intermediate_size == 0 {
    return Err("intermediate_size must be > 0".into());
}
if self.local_attention == 0 {
    return Err("local_attention must be > 0".into());
}
if !self.layer_norm_eps.is_finite() || self.layer_norm_eps <= 0.0 {
    return Err("layer_norm_eps must be finite and > 0".into());
}
```

`num_hidden_layers` の i32 bound は不要 (model.rs:236 の range iterator でしか使われず、i32 キャストされる箇所がない)。テストは config/tests.rs に4ケース追加。

## 規模

S (validate への条件追加 + テスト4件)

## 出典

goal-driven code review session 2026-06-10 で表出。
